### PR TITLE
Two related fixes for NgModule compilation output

### PR DESCRIPTION
**Pull request by alxhub**

### CODE REVIEW

File: packages/core/src/render3/jit/module.ts [hunk 1]
```diff
@@ -125,7 +125,6 @@ export function compileNgModuleDefs(
               exports: flatten(ngModule.exports || EMPTY_ARRAY)
                            .map(resolveForwardRef)
                            .map(expandModuleWithProviders),
-              emitInline: true,
```
Comment (petebacondarwin) [inline packages/core/src/render3/jit/module.ts]: In the jit_compiler_facade you replaced this with `guardJitDefinition: false,`. Why not here?

**Review by petebacondarwin — APPROVED** [review]
A couple of nits. Also I didn't understand this sentence in the commit message:

```
2) it breaks tree-shaking to include, as it describes the full module scope
   graph (every declaration of every module in the app)
```

File: packages/compiler-cli/src/ngtsc/translator/src/translator.ts [hunk 2]
```diff
@@ -251,16 +251,25 @@ class ExpressionTranslatorVisitor implements ExpressionVisitor, StatementVisitor
...
       throw new Error(`Import unknown module or symbol ${ast.value}`);
     }
-    const {moduleImport, symbol} =
-        this.imports.generateNamedImport(ast.value.moduleName, ast.value.name);
-    if (moduleImport === null) {
-      return ts.createIdentifier(symbol);
+    // If a moduleName is specified, this is a normal import. If there's no module name, it's a
+    // reference to a global/ambient symbol.
+    if (ast.value.moduleName !== null) {
+      // THis is a normal import. Find the imported module.
```
[the remark this answers is not in the document]
Comment (alxhub) [reply packages/compiler-cli/src/ngtsc/translator/src/translator.ts]: Fixed.

File: packages/compiler/src/render3/r3_module_compiler.ts [hunk 3]
```diff
@@ -84,50 +82,47 @@ export function compileNgModule(meta: R3NgModuleMetadata): R3NgModuleDef {
...
+    exports?: o.Expression,
+  }|null = null;
+
   // If requested to emit scope information inline, pass the declarations, imports and exports to
   // the `ɵɵdefineNgModule` call. The JIT compilation uses this.
-  if (emitInline) {
-    if (declarations.length) {
-      definitionMap.declarations = refsToArray(declarations, containsForwardDecls);
-    }
+  if (declarations.length > 0 || imports.length > 0 || exports.length > 0) {
```
[the remark this answers is not in the document]
Comment (alxhub) [reply packages/compiler/src/render3/r3_module_compiler.ts]: Removed, thanks.

File: packages/core/src/render3/jit/module.ts (hunk 1 above)
Comment (alxhub) [reply packages/core/src/render3/jit/module.ts]: This is a JIT-specific interface, so there's no need to expose the `guardJitDefinition` switch. It's _always_ going to be `false`.

**Review by IgorMinar — CHANGES_REQUESTED** [review]
this PR is a bit confusing because it has two unrelated parts.

- the IIFE is valuable as it removes the burden from build-optimizer to do this kind of rewrite. 
  - this change is all good, and will require a non-critical cleanup change in build-optimizer to remove the pass that adds this IIFE wrapper during build optimizations. it's great that we can do this during codegen and not have to correct the issue during each prod build. Can you please create a backlog item for the framework team? // heads up @filipesilva 
- the second part is all about supporting jit mode with prod optimizations, or rather ensuring that prod optimizations don't break the jit mode.
  - this change will require a modification in CLI to support the new terser flag array, as well as closure compiler config change in google3 to define `ngJitMode` variable.

**Review by IgorMinar — COMMENTED** [review]
btw I don't think that the ngModuleDef change is a feature, I think that this is a fix, because without it the JIT with prod optimizations is broken

File: packages/compiler-cli/src/tooling.ts [hunk 4]
```diff
@@ -20,3 +20,8 @@ export const GLOBAL_DEFS_FOR_TERSER = {
   ngDevMode: false,
   ngI18nClosureMode: false,
 };
+
+export const GLOBAL_DEFS_FOR_TERSER_WITH_AOT = {
+  ...GLOBAL_DEFS_FOR_TERSER,
+  ngJitMode: false,
```
[the remark this answers is not in the document]
Comment (alxhub) [reply packages/compiler-cli/src/tooling.ts]: The CLI already reads this if present: https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts#L362-L372

[the remark this answers is not in the document]
Comment (IgorMinar) [reply packages/compiler-cli/src/tooling.ts]: aha! I did not know that. In that case CLI is good to go. thanks for the info.

**Review by alan-agius4 — COMMENTED** [review]
Do we except users that can’t use AOT to provide set `ngJitMode` globally?

Example;
System JS users
Codepen/plunkr etc...

**Edit: Never mind misread the statement 😥**

**Pull request closed without merging**